### DISCUSSION TITLE
Ability to run Powershell commands after clicking a button?

### DISCUSSION THREAD

**Discussion by mckayc** [Windos/BurntToast#90] [General]
So I just discovered BurntToast yesterday and am excited to use it!

Here is the situation I am trying to solve; at our company we often run Powershell scripts via GPO to install software or make changes to computers. One issue with this is that we would like to notify employees before we install software and give them an option to postpone.

The alarm clock function seems to be perfect for this.

`New-BurntToastNotification -Text 'WAKE UP!' -Sound 'Alarm2' -SnoozeAndDismiss`

You could essentially keep this as is with a Snooze button and replace the wording of "Dismiss" with "Run" then and people click the run button it runs the Powershell script.

Is something like this already possible?

Thanks!

(PS I realize this is not really an issue per se but I don't know where else to post this. If there is another place I can post this questions please let me know)

**Comment by Windos** [comment]
Hey @mckayc, unfortunately this isn't possible at the moment mainly because of limitation where PowerShell can't register actions against a "Windows RT" events.

There's an [issue](https://github.com/PowerShell/PowerShell/issues/2181) open about this on the PowerShell repo which has had some action recently.

Unfortunately, unless someone creates a wrapper/shim external to PowerShell any fixes that enable this will likely only apply to newer versions of PowerShell (e.g. 7.1+) meaning buttons that perform an action will require a PowerShell upgrade.

This is 100% the holy grail for toasts... I'm *very* invested in getting it working because it opens up *so* many doors.

**Comment by mckayc** [comment]
Thanks for your reply! Good to know what currently is and is not possible.

**Comment by jfthuong** [comment]
We could do something like:
1. Powershell main script registers a custom URI **powershell** to launch a given function (via the registry)
2. Assign the URI to the button

The URI could be something like `powershell://path-to-my-script.ps1#my-function?arg1=value1&arg2=value2`.

I remember seeing this somewhere in one of the talks about BurntToast.

If that is a viable option, we could have BurntToast that is doing actions behind the scene:
* Registering a specific URI (if not done already)
* Providing a more convenient way to assign the function to the button (a function like `New-BTButtonAction` or something like that)

Just some thoughts....

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
